examples/1.0.0/bnpl-arazzo.yaml

```diff
@@ -0,0 +1,252 @@
+
+      If the customer is eligible for the BNPL loan then a customer resource is created 
+    operationId: createCustomer
+    parameters:
+      - name: customerAttributes
+        in: body
+        value: $inputs.customer
+      - name: termsAndConditionsAccepted
+        in: body
+        value: true
+    successCriteria:
+      - condition: $statusCode == 200
+      - condition: $statusCode == 201
+    onSuccess:
+      - name: customerIsEligible
+        type: goto
+        stepId: initiateBnplTransaction
+        criteria:
+          - condition: $statusCode == 201
+      - name: customerIsNotEligible
+        type: end
+        criteria:
+          - condition: $statusCode == 200
+    outputs:
+      customer: $response.body.links.self
+  - stepId: initiateBnplTransaction
+    description: Initiate the BNPL transaction by sending the customer identifier, eligible products, and indicative loan amount to initiate the loan process
+    operationId: createBnplTransaction
+    parameters:
+      - name: enrolledCustomer
+        in: body
+        value: $inputs.customer.uri
+      - name: newCustomer
+        in: body
+        value: $steps.createCustomer.outputs.customer
+      - name: products
+        in: body
+        value: $steps.checkLoanCanBeProvided.outputs.eligibleProducts
+    successCriteria:
+      - condition: $statusCode == 202
+    onSuccess:
+      - name: CustomerAuthorizationRequired
+        type: goto
+        stepId: authenticateCustomerAndAuthorizeLoan
+        condition: $response.body.redirectAuthToken != null
+      - name: CustomerAuthorizationNotRequired
+        type: goto
+        stepId: retrieveFinalizedPaymentPlan
+        condition: $response.body.redirectAuthToken == null
+    outputs:
+      redirectAuthToken: $response.body.redirectAuthToken
+      loanTransactionResourceUrl: $response.body.links.self
+  - stepId: authenticateCustomerAndAuthorizeLoan
+    description: |
+      Authenticate the customer and seek authorization for the loan.
+
+      Notes:
+      
+      - Authenticate in this case does not necessarily mean with a valid set of credentials. It could be to prove the identity of the end user, possibly doing an UMA style interaction (which is too complex to tie into this example).
+      - A redirect is returned for to be sent to the customer. The customer should follow this, but the success of the authorisation is out-of-band.
+      - This flow mimics OAuth style authorization but is not an exact match as it does not rely on an authorisation code/access token exchange (as this seems somewhat artificial in this context).
+
+    operationId: getAuthorization
+    parameters:
+      - name: redirectAuthToken
+        in: query
+        value: $steps.authenticateCustomerAndAuthorizeLoan.outputs.redirectAuthToken
+    successCriteria:
+      - condition: $statusCode == 302
+    outputs:
+      redirectUrl: $response.headers.Location
+  - stepId: retrieveFinalizedPaymentPlan
```

Comment:
Should the payment plan be return in an output variable....even perhaps as part of the output of the workflow?

Reply:
Added `step` output and Workflow output as suggested